scripts/generateCliCode.ts

```diff
@@ -0,0 +1,356 @@
+      return this.collectServerSettingsForJSONProperty(propertyName, settingsTree[propertyName], includeJSONTag, indent);
+    });
+  }
+
+  protected collectServerSettingsForJSONProperty(propertyName: string, typeWrapper: settingsTypeObject, includeJSONTag: boolean, indent: string): string[] {
+    if (typeof (typeWrapper.type) === 'object') {
+      const lines: string[] = [];
+
+      lines.push(`${ indent }${ capitalize(propertyName) } struct {`);
+      lines.push(...this.collectServerSettingsForJSON(typeWrapper.type, includeJSONTag, `${ indent }  `));
+      const lastLineParts = [indent, '}'];
+
+      if (includeJSONTag) {
+        lastLineParts.push(` \`json:"${ propertyName }"\``);
+      }
+      lines.push(lastLineParts.join(''));
+
+      return lines;
+    } else {
+      const onlyLineParts = [indent, capitalize(propertyName), ' '];
+
+      if (includeJSONTag) {
+        onlyLineParts.push('*');
+      }
+      onlyLineParts.push(typeWrapper.type);
+      if (includeJSONTag) {
+        onlyLineParts.push(`\`json:"${ propertyName },omitempty"\``);
+      }
+
+      return [onlyLineParts.join('')];
+    }
+  }
+
+  protected convertStringsToGolang(enums: string[] | undefined): string {
+    return !enums ? '' : `[]string{${ enums.map(s => JSON.stringify(s) ).join(', ') }}`;
+  }
+
+  protected getCommandLineArgValue(flagType: goCmdFlagTypeName, capitalizedName: string) {
+    switch (flagType) {
+    case 'Bool':
+      return `+"="+strconv.FormatBool(specifiedSettings.${ capitalizedName })`;
+    case 'Int':
+      return `, strconv.Itoa(specifiedSettings.${ capitalizedName })`;
+    case 'String':
+      return `, specifiedSettings.${ capitalizedName }`;
+    }
+  }
+
+  protected getFullUsageNote(usageNote: string, rawEnums: undefined|string[]): string {
+    const usageParts = [usageNote];
+
+    if (rawEnums) {
+      usageParts.push(`(Allowed values: [${ rawEnums.join(', ' ) }].)`);
+    }
+
+    return usageParts.join(' ').trim();
+  }
+
+  protected updateLeaf(propertyName: string, capitalizedName: string,
+    lcTypeName: goTypeName, flagType: goCmdFlagTypeName,
+    defaultValue: string, preference: yamlObject,
+    settingsTree: settingsTreeType) {
+    const enums = this.convertStringsToGolang(preference.enum);
+    const usageNote = preference['x-rd-usage'] ?? '';
+    const newFlag: commandFlagType = {
+      capitalizedName,
+      defaultValue,
+      flagType,
+      propertyName,
+      enums,
+      aliasFor:  '',
+      valuePart: this.getCommandLineArgValue(flagType, capitalizedName),
+    };
+
+    newFlag.usageNote = this.getFullUsageNote(usageNote, preference.enum);
+    settingsTree[lastName(propertyName)] = { type: lcTypeName };
+    this.commandFlags.push(newFlag);
+    for (const alias of preference['x-rd-aliases'] ?? []) {
+      const aliasFlag = Object.assign({}, newFlag, { propertyName: alias, aliasFor: propertyName });
+
+      aliasFlag.usageNote = this.getFullUsageNote(usageNote, preference.enum);
```

Comment:
Isn't this exactly the same as the value that's already there (because `aliasFlag` is based on the `newFlag` that already has a `usageNote`)? A previous version of the PR added something about it being an alias for a different command, but that no longer exists in this version.

Reply:
Yes, exactly. Good catch.